lib/recipe.rb

```diff
@@ -0,0 +1,29 @@
+class Recipe
+  attr_reader :name, :ingredients_required
+
+  def initialize(name)
+    @name = name 
+    @ingredients_required = {}
+  end 
+
+  def add_ingredient(ingredient, quantity)
+    if @ingredients_required[ingredient].nil?
+      @ingredients_required[ingredient] = quantity
+    else 
+      @ingredients_required[ingredient] += quantity
+    end
+  end
+
+  def ingredients
+    @ingredients_required.select do |ingredient, quantity|
```

Comment:
hmm, so with .select, it's looking for the block to be truthy, which ingredient will be, so this will return the original @ingredients_required hash, then you're calling .keys on that, so you could just have this method do @ingredients_requried.keys.